compiler/sigmatch.nim

```diff
@@ -1528,7 +1528,7 @@ proc typeRel(c: var TCandidate, f, aOrig: PType,
         reduceToBase(a)
     if effectiveArgType.kind == tyObject:
       if sameObjectTypes(f, effectiveArgType):
-        c.inheritancePenalty = 0
+        c.inheritancePenalty = if tfInheritable in f.flags: 0 else: -1
```

Comment:
`tfInheritable` isn't always defined for objects with inheritance, but `tfFinal` is for objects without inheritance. Checking this instead should work.